kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/DefaultProducerFactory.java

```diff
@@ -273,13 +293,25 @@ public void close() {
         public void close(long timeout, TimeUnit unit) {
             boolean isAdded = this.cache.offer(this);
             if (!isAdded) {
-                this.delegate.close(timeout, unit);
+                super.close(timeout, unit);
             }
         }
+    }
+
+    private static final class CloseNeverProducer<K, V> extends ProducerDecorator<K, V> {
+
+        CloseNeverProducer(Producer<K, V> delegate) {
```

Comment:
Wouldn't this mean you'd have a dangling `Producer` at all times? Do you care to elaborate what would happen on a shutdown in this scenario?

Reply:
On a shutdown, the `DefaultProducerFactory` explicitly calls `delegate.close()` and thus bypasses the no-op implementation in the decorator.

Reply:
Ah darn of course, thanks for pointing this  out @lbilger.